packages/mds-ingest-service/repository/index.ts

```diff
@@ -14,14 +14,226 @@
  * limitations under the License.
  */
 
-import { ReadWriteRepository } from '@mds-core/mds-repository'
+import { InsertReturning, ReadWriteRepository, RepositoryError } from '@mds-core/mds-repository'
+import { Enum, PROPULSION_TYPE, Timestamp, UUID, VEHICLE_EVENT, VEHICLE_STATE, VEHICLE_TYPE } from '@mds-core/mds-types'
+import { isUUID } from '@mds-core/mds-utils'
+import {
+  EventDomainModel,
+  EventDomainCreateModel,
+  TelemetryDomainCreateModel,
+  DeviceDomainCreateModel
+} from '../@types'
 import entities from './entities'
+import { DeviceEntity } from './entities/device-entity'
+import { EventEntity } from './entities/event-entity'
+import { TelemetryEntity } from './entities/telemetry-entity'
+import {
+  DeviceDomainToEntityCreate,
+  DeviceEntityToDomain,
+  EventDomainToEntityCreate,
+  EventEntityToDomain,
+  TelemetryDomainToEntityCreate,
+  TelemetryEntityToDomain
+} from './mappers'
 import migrations from './migrations'
 
+/**
+ * Aborts execution if not running under a test environment.
+ */
+const testEnvSafeguard = () => {
+  if (process.env.NODE_ENV !== 'test') {
+    throw new Error(`This method is only supported when executing tests`)
+  }
+}
+
+const GROUPING_TYPES = Enum('latest_per_vehicle', 'latest_per_trip', 'all_events')
+export type GROUPING_TYPE = keyof typeof GROUPING_TYPES
+
+export type TimeRange = {
+  start: Timestamp
+  end: Timestamp
+}
+export interface GetVehicleEventsFilterParams {
+  vehicle_types?: VEHICLE_TYPE[]
+  propulsion_types?: PROPULSION_TYPE[]
+  provider_ids?: UUID[]
+  vehicle_states?: VEHICLE_STATE[]
+  time_range: TimeRange
+  grouping_type: GROUPING_TYPE
+  device_or_vehicle_id?: string // Match on device_id or vehicle_id
+  device_ids?: UUID[]
+  event_types?: VEHICLE_EVENT[]
+  geography_ids?: UUID[]
+}
 class IngestReadWriteRepository extends ReadWriteRepository {
   constructor() {
     super('ingest', { entities, migrations })
   }
+
+  public createEvents = async (events: EventDomainCreateModel[]) => {
```

Comment:
Should we add support for chunked inserts to any of these `create` methods?

Reply:
Not yet, no use case for this volume ATM.